packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -108,16 +126,88 @@ jni::local_ref<JCxxCallbackImpl::JavaPart> createJavaCallbackFromJSIFunction(
                     jsi::valueFromDynamic(strongWrapper2->runtime(), val));
               }
 
-              strongWrapper2->callback().call(
-                  strongWrapper2->runtime(),
-                  (const jsi::Value *)args.data(),
-                  args.size());
+              callbackWrapperExecutor(strongWrapper2, args);
             });
 
         wrapperWasCalled = true;
       });
 }
 
+jni::local_ref<JCxxCallbackImpl::JavaPart> createJavaCallbackFromJSIFunction(
+    jsi::Function &&function,
+    jsi::Runtime &rt,
+    const std::shared_ptr<CallInvoker> &jsInvoker) {
+  auto executor = [](
+          const std::shared_ptr<CallbackWrapper>& wrapper,
+          const std::vector<jsi::Value>& args) {
+      wrapper->callback().call(
+              wrapper->runtime(),
+              (const jsi::Value *)args.data(),
+              args.size());
+  };
+  return createJavaCallbackFromJSIFunction(std::move(function), rt, jsInvoker, std::move(executor));
+}
+
+jsi::Value createJSRuntimeError(jsi::Runtime &runtime, const std::string &message);
+
+jni::local_ref<JCxxCallbackImpl::JavaPart> createPromiseRejectJavaCallbackFromJSIFunction(
+    const std::weak_ptr<CallbackWrapper> &function,
+    jsi::Runtime &rt,
+    const std::shared_ptr<CallInvoker> &jsInvoker,
+    std::optional<std::string> jsInvocationStack) {
+  auto executor = [jsInvocationStack = std::move(jsInvocationStack)](
+          const std::shared_ptr<CallbackWrapper>& reject,
+          const std::vector<jsi::Value>& args) {
+      std::string message;
+      jsi::Runtime &rt2 = reject->runtime();
+      const jsi::Value* cause = args.data();
```

Comment:
You should validate the length of args.

```suggestion
     jsi::Value &cause = args[0];
```